src/client/java/teammates/client/scripts/DataMigrationForChangeOfGoogleIdInITESM.java

```diff
@@ -0,0 +1,141 @@
+package teammates.client.scripts;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.tools.cloudstorage.GcsFilename;
+import com.google.appengine.tools.cloudstorage.GcsService;
+import com.google.appengine.tools.cloudstorage.GcsServiceFactory;
+import com.google.appengine.tools.cloudstorage.RetryParams;
+import com.googlecode.objectify.Key;
+import com.googlecode.objectify.cmd.Query;
+
+import teammates.common.util.Config;
+import teammates.storage.entity.Account;
+import teammates.storage.entity.CourseStudent;
+import teammates.storage.entity.Instructor;
+import teammates.storage.entity.StudentProfile;
+import teammates.test.driver.TestProperties;
+
+/**
+ * Data migration to change google ID as requested by ITESM.
+ */
+public class DataMigrationForChangeOfGoogleIdInITESM extends DataMigrationEntitiesBaseScript<Account> {
+
+    private static final Pattern STUDENT_GOOGLE_ID_PATTERN = Pattern.compile("A\\d{8}@itesm\\.mx");
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForChangeOfGoogleIdInITESM().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<Account> getFilterQuery() {
+        return ofy().load().type(Account.class);
+    }
+
+    @Override
+    protected boolean isPreview() {
+        return true;
+    }
+
+    @Override
+    protected String getLastPositionOfCursor() {
+        return "";
+    }
+
+    @Override
+    protected int getCursorInformationPrintCycle() {
+        return 100;
+    }
+
+    @Override
+    protected boolean shouldUseTransaction() {
+        // queries for student & instructor entities are not supported by the Datastore
+        return false;
+    }
+
+    @Override
+    protected boolean isMigrationNeeded(Key<Account> key) throws Exception {
+        Account account = ofy().load().key(key).now();
+
+        if (!account.isInstructor()) {
+            // note that only ITESM instructors have been migrated to tec.mx
+            return false;
+        }
+
+        String googleId = account.getGoogleId();
+        if (!googleId.endsWith("@itesm.mx")
+                || STUDENT_GOOGLE_ID_PATTERN.matcher(googleId).matches()) {
+            // only in ITESM institute
+            // the googleId should NOT be of student pattern
+            return false;
+        }
+
+        println(String.format("Going to migrate googleId %s", googleId));
```

Comment:
1. migrate **account with** googleId
1. You might want to add the new googleId here as well, otherwise you can't know whether the new googleId is what is required.